lib/text.rb

```diff
@@ -0,0 +1,21 @@
+class Text
+
+  attr_reader :time
+
+  def initialize(t = Time)
+    @t = t
+    @time = nil
+  end 
+
+  def text(_order)
+    delivery_time
+    "Thank you! Your order was placed and will be delivered before #{time}"
+  end
+
+  private 
+
+  def delivery_time
+    @t = @t.new + 1 * 60 * 60
+    @time = @t.strftime("%H:%M")
+  end
```

Comment:
Again, it is better to return a value than to store it in an instance variable.

```rb
def text
  delivery_time = calculate_delivery_time(Time.new)
  "Thank you! Your order was placed and will be delivered before #{delivery_time}"
end
```

This way you can remove both the `@t` and `@time` instance variables.